manifests/init.pp

```diff
@@ -151,30 +151,34 @@
   ### END Service Configuration ###
 
   ### START Hiera Lookups ###
-  $geo_mappings                                     = {},
-  $string_mappings                                  = {},
-  $nginx_locations                                  = {},
-  $nginx_locations_defaults                         = {},
-  $nginx_mailhosts                                  = {},
-  $nginx_mailhosts_defaults                         = {},
-  $nginx_streamhosts                                = {},
-  $nginx_upstreams                                  = {},
-  $nginx_servers                                    = {},
-  $nginx_servers_defaults                           = {},
+  Hash $geo_mappings                                = {},
+  Hash $geo_mappings_defaults                       = {},
+  Hash $nginx_locations                             = {},
+  Hash $nginx_locations_defaults                    = {},
+  Hash $nginx_mailhosts                             = {},
+  Hash $nginx_mailhosts_defaults                    = {},
+  Hash $nginx_servers                               = {},
+  Hash $nginx_servers_defaults                      = {},
+  Hash $nginx_streamhosts                           = {},
+  Hash $nginx_streamhosts_defaults                  = {},
+  Hash $nginx_upstreams                             = {},
+  Hash $nginx_upstreams_defaults                    = {},
+  Hash $string_mappings                             = {},
+  Hash $string_mappings_defaults                    = {},
   ### END Hiera Lookups ###
 ) inherits ::nginx::params {
 
   contain '::nginx::package'
   contain '::nginx::config'
   contain '::nginx::service'
 
-  create_resources('nginx::resource::upstream', $nginx_upstreams)
-  create_resources('nginx::resource::server', $nginx_servers, $nginx_servers_defaults)
-  create_resources('nginx::resource::location', $nginx_locations, $nginx_locations_defaults)
-  create_resources('nginx::resource::mailhost', $nginx_mailhosts, $nginx_mailhosts_defaults)
-  create_resources('nginx::resource::streamhost', $nginx_streamhosts)
-  create_resources('nginx::resource::map', $string_mappings)
-  create_resources('nginx::resource::geo', $geo_mappings)
+  create_resources( 'nginx::resource::geo', $geo_mappings, $geo_mappings_defaults )
```

Comment:
Isn't this now considered bad practice? Instead we should loop over the hash. As an example: https://github.com/bastelfreak/puppet-ferm/blob/master/manifests/init.pp#L48

CC: @hunner @eputnam. What is your opinion here? We should update our review guidelines I guess.

Reply:
wouldn't this change imply another request due to its nature ?

Reply:
yes, let's keep it as is.